Wikipedia/Code/WMFArticleFooterMenuViewController.m

```diff
@@ -100,10 +100,19 @@ - (void)tableView:(UITableView*)tableView didSelectRowAtIndexPath:(NSIndexPath*)
 - (void)showDisambiguationItems {
     WMFDisambiguationPagesViewController* articleListVC = [[WMFDisambiguationPagesViewController alloc] initWithArticle:self.article dataStore:self.dataStore];
     articleListVC.title = MWLocalizedString(@"page-similar-titles", nil);
-    [self presentViewController:[[UINavigationController alloc] initWithRootViewController:articleListVC] animated:YES completion:^{
+    UINavigationController *navController = [[UINavigationController alloc] initWithRootViewController:articleListVC];
+    navController.delegate = self;
+    [self presentViewController:navController animated:YES completion:^{
     }];
 }
 
+- (void)navigationController:(UINavigationController*)navigationController
+      willShowViewController:(UIViewController*)viewController
+                    animated:(BOOL)animated {
+    BOOL isToolbarEmpty = [viewController toolbarItems].count == 0;
+    [navigationController setToolbarHidden:isToolbarEmpty];
```

Comment:
this is identical to the method in `WMFAppViewController`. Would you please DRY the "hide toolbar if empty" logic into a category instead of copy/pasting?